crates/services/upgradable-executor/src/executor.rs

```diff
@@ -383,7 +386,12 @@ where
                     self.native_produce_inner(block, options, dry_run)
                 }
                 ExecutionStrategy::Wasm { module } => {
-                    self.wasm_produce_inner(module, block, options, dry_run)
+                    let maybe_blocks_module = self.get_module(block_version);
```

Comment:
It is a bug, that we used `WASM_BYTECODE` from the binary itself instead of using value from the database(if we have any). For example, if you run the node with `FUEL_ALWAYS_USE_WASM`, it will use the built-in(into `fuel-core` binary) WASM byte code instead of the byte code that comes with the upgrade of the network. If you compiled your binary with wrong features flag, it can stop you form syncing with the network even with WASM executor